File: app/src/main/java/com/example/pam_project/networking/authors/AuthorsMapper.java
package com.example.pam_project.networking.authors;

import java.util.ArrayList;
import java.util.List;

public class AuthorsMapper {

    public List<AuthorsModel> map(final ListAuthorsResponse response) {
        List<AuthorsModel> authors = new ArrayList<>();

        for (int i = 0; i < response.allAuthors.size(); i++) {
            AuthorsResponse currentAuthor = response.allAuthors.get(i);
            if (currentAuthor.lastName == null) {
                throw new IllegalArgumentException("Lastname was expected");
            }

            AuthorsModel author;

            if (currentAuthor.firstName == null) {
                author = new AuthorsModel(currentAuthor.lastName);
            } else {
                author = new AuthorsModel(currentAuthor.firstName, currentAuthor.lastName);
            }

            authors.add(author);
        }

        return authors;
    }
}

File: app/src/main/java/com/example/pam_project/networking/authors/AuthorsMapper.kt
package com.example.pam_project.networking.authors

import java.util.*

class AuthorsMapper {
    fun map(response: ListAuthorsResponse?): List<AuthorsModel> {
        val authors: MutableList<AuthorsModel> = ArrayList()
        for (i in response!!.allAuthors.indices) {
            val currentAuthor = response.allAuthors[i]
            requireNotNull(currentAuthor!!.lastName) { "Lastname was expected" }
            val author: AuthorsModel = if (currentAuthor.firstName == null) {
                AuthorsModel(currentAuthor.lastName)
            } else {
                AuthorsModel(currentAuthor.firstName, currentAuthor.lastName)
            }
            authors.add(author)
        }
        return authors
    }
}